### PR TITLE
add error information about sqlite keywords usage

### PR DESCRIPTION
**Pull request by geracervantes**

### CODE REVIEW

**Comment by bpappin** [comment]
Just a couple of suggestions:

1. It might be easier to maintain as an array of keywords instead of a list. Since it only runs when the generator runs, and is immutable, the collection is a bit overkill.
2. in createTableSQL(), you should output the name of the keyword in the error so the developer can easily figure out which column has a bad name.

File: library/src/main/java/com/orm/KeyWords.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        reservedWords.add("VACUUM");
+        reservedWords.add("VALUES");
+        reservedWords.add("VIEW");
+        reservedWords.add("VIRTUAL");
+        reservedWords.add("WHEN");
+        reservedWords.add("WHERE");
+        reservedWords.add("WITH");
+        reservedWords.add("WITHOUT");}
+
+    public boolean isaReservedWords(String word){
```
Comment (sibeliusseraphini) [inline library/src/main/java/com/orm/KeyWords.java]: this should be static

File: library/src/main/java/com/orm/SchemaGenerator.java [hunk 2]
```diff
@@ -177,9 +177,14 @@ private void addColumns(Class<?> table, SQLiteDatabase sqLiteDatabase) {
     }
 
     protected String createTableSQL(Class<?> table) {
+        KeyWords link = new KeyWords();
         Log.i(SUGAR, "Create table if not exists");
         List<Field> fields = ReflectionUtil.getTableFields(table);
         String tableName = NamingHelper.toSQLName(table);
+        if(link.isaReservedWords(tableName))
```
Comment (sibeliusseraphini) [inline library/src/main/java/com/orm/SchemaGenerator.java]: use a static method for this

File: library/src/main/java/com/orm/SchemaGenerator.java [hunk 3]
```diff
@@ -177,9 +177,14 @@ private void addColumns(Class<?> table, SQLiteDatabase sqLiteDatabase) {
     }
 
     protected String createTableSQL(Class<?> table) {
+        KeyWords link = new KeyWords();
```
Comment (sibeliusseraphini) [inline library/src/main/java/com/orm/SchemaGenerator.java]: it does not make sense to create an object of KeyWords

**Comment by sibeliusseraphini** [comment]
@geracervantes or @JonatanSalas,  could someone change from Set to an Array

> array of keywords instead of a list

and also address this:

> in createTableSQL(), you should output the name of the keyword in the error so the developer can easily figure out which column has a bad name.

and use  static method instead of creating a KeyWords object

**Comment by geracervantes** [comment]
working on it

**Pull request merged**